### PR TITLE
Adding method to get endpoint URL

### PR DESCRIPTION
**Pull request by thushara35**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 1]
```diff
@@ -4285,6 +4285,38 @@ public static String getAuthServerURL() throws APIManagementException {
...
+     * @return endpoints comma separated string
+     * @throws APIManagementException If API key manager URL unspecified
+     */
+    public static String getGatewayEndpoints(API api) throws APIManagementException {
+        String endpoints = "none";
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
+                getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        String publishedEnvironments = writeEnvironmentsToArtifact(api);
+        String[] publishedEnvironmentSet = publishedEnvironments.split(",");
+        for (int i = 0; i < config.getApiGatewayEnvironments().keySet().toArray().length; i++) {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Gateway environments return a list right. YO

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 2]
```diff
@@ -4285,6 +4285,38 @@ public static String getAuthServerURL() throws APIManagementException {
...
+            for (int j = 0; j < publishedEnvironmentSet.length; j++) {
+                if (config.getApiGatewayEnvironments().keySet().toArray()[i].toString()
+                        .equals(publishedEnvironmentSet[j])) {
+                    Environment environment = (Environment) config.getApiGatewayEnvironments().values().toArray()[i];
+                    endpoints = environment.getApiGatewayEndpoint();
+                    break;
+                }
+            }
+        }
+        if (StringUtils.isBlank(publishedEnvironments)) {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: shouldn't these exception throwing come before the for loop. so why should be return exception, can we simply send there is no gateway endpoints or similar thing without throwing exception :sunglasses:

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 1 above)
Comment (thushara35) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: yes, Fixed it :)

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 2 above)
Comment (thushara35) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Yes this is redundant. Will remove the first exception

**Pull request merged**